Update IPv4/IPv6 installation prerequisites (#3545)
theforeman/foreman-documentation: guides/common/modules/con_ipv4-and-ipv6-requirements.adoc

[id="ipv4-and-ipv6-requirements_{context}"]
= IPv4 and IPv6 requirements

You can install {Project} in an IPv4 network or in an IPv6 network.

ifdef::satellite[]
[NOTE]
====
Dual-stack {Project} installation that uses both IPv4 and IPv6 is not supported.
====
endif::[]

The following requirements apply to installations in an IPv4 network:

* Ensure that an IPv6 loopback is configured.
* Do not disable IPv6 in kernel by adding the `ipv6.disable=1` kernel parameter.
ifdef::satellite[]
+
For a supported way to disable the IPv6 protocol, see link:https://access.redhat.com/solutions/5045841[How do I disable the IPv6 protocol on Red Hat Satellite and/or Red Hat Capsule server?] in Red{nbsp}Hat Knowledgebase.
endif::[]

> The following requirements apply to installations in an IPv6 network:

Reviewer: There are quite a few specific provisioning limitations. I wonder if we should mention those in the installation guide already or instead mention them in the provisioning guide. Then include a note here that there are provisioning specific limitation.

Reviewer: True, but on the other hand it's also good to remind admins who set up provisioning because they may not remember from installing it or it changed since. That's why I think it's good to have it at least in the provisioning guide.

Reviewer 2: To decide which would work best here, one would need to take a look at the requirement list here and at the provisioning guide. That's not something I should get involved in right now but since Avital is planning a redesign of the installation guide, I summed up this comment thread in her Jira epic: https://issues.redhat.com/browse/SAT-23920

> * You must deploy an external DHCP IPv6 server as a separate, unmanaged service to bootstrap clients into GRUB2.

Reviewer: This is only true if you provision. If you don't there's no requirement to deploy this.

> GRUB2 configures IPv6 networking by using DHCPv6 or by assigning a static IPv6 address.

Reviewer: I think GRUB can also do SLAAC (stateless IPv6 address assignments). A problem is that we can't provide the required information where to boot from.

Taking a step back, there are 3 ways to configure IPv6 addresses:
* Static: user provides this info manually
* SLAAC: https://en.wikipedia.org/wiki/IPv6#Stateless_address_autoconfiguration_(SLAAC)
* DHCPv6

SLAAC can only provide a limited amount of information. It's sufficient to provide addresses and DNS, but in particular for provisioning, it can't provide info like the TFTP server address and filename or HTTPBoot URL which we need.

That information is crucial because on fresh installations you don't have anything. We download GRUB (or something else) from a TFTP or HTTPBoot location.

> This is required because the DHCP server in {RHEL} (ISC DHCP) does not provide an integration API for managing IPv6 records, therefore the {SmartProxy} DHCP plugin that provides DHCP management is limited to IPv4 subnets.

Reviewer: Preempting customer questions: Infoblox does have an API to manage DHCPv6. Can I use that? The answer is no, so perhaps just reduce it to say we currently can't manage DHCPv6?

ifdef::satellite[]
* You must deploy an external HTTP proxy server that supports both IPv4 and IPv6.
This is required because Red Hat Content Delivery Network distributes content only over IPv4 networks, therefore you must use this proxy to pull content into the {Project} on your IPv6 network.
endif::[]
ifndef::satellite[]

> * Optional: If you rely on content from IPv4 networks, you must deploy an external IPv4 HTTP proxy server.
> This is required to access Content Delivery Networks that distribute content only over IPv4 networks, therefore you must use this proxy to pull content into {Project} on your IPv6 network.

Reviewer: This feels redundant: if you rely on X, you must .... This is required to X.

Also, in upstream we don't have a note that dual stack networks aren't supported so this can work just fine.

Perhaps reduce it to something like:
```suggestion
* Optional: If you rely on content from IPv4-only networks and are deployed on IPv6-only, you can use an external dual stack HTTP proxy server.
```

endif::[]
+

> You must configure {Project} to use this dual-stack (supporting both IPv4 and IPv6) HTTP proxy server as the default proxy.

Reviewer: For non-Satellite you it previously said `Optional` and now it's `must`.

For more information, see {InstallingServerDocURL}adding-a-default-http-proxy_{project-context}[Adding a Default HTTP Proxy to {Project}].

Although {Project} provisioning templates include IPv6 support for PXE and HTTP (iPXE) provisioning, the only tested and certified provisioning workflow is the UEFI HTTP Boot provisioning.

> This limitation only relates to users who plan to use {Project} to provision hosts.

Reviewer 2: @Lennonka Is this relevant for you with regards to https://github.com/theforeman/foreman-documentation/pull/3532? I'm letting you know just in case it's something you want to look into as part of your PR. (These two lines are currently in the installation guide, my PR doesn't add them.)

Reviewer 3: Thank you for letting me know! Yes, that will need an update. I'll take a look at it once your PR will have been merged.

endif::[]

The following requirements apply to installations in an IPv6 network:

* If you intend to use {Project} to provision hosts, you must deploy an external DHCPv6 server and cofigure it manually to communicate with the network boot process and to manage IP address assignment because {Project} cannot integrate with a DHCPv6 server and manage its configuration.
For more information about DHCPv6 server configuration, see {ProvisioningDocURL}options-in-unmanaged-dhcpv6[Options in unmanaged DHCPv6] in _{ProvisioningDocTitle}_.
ifdef::satellite[]
* You must deploy an external HTTP proxy server that supports both IPv4 and IPv6.
This is required because Red Hat Content Delivery Network distributes content only over IPv4 networks, therefore you must use this proxy to pull content into the {Project} on your IPv6 network.
You must configure {Project} to use this dual-stack (supporting both IPv4 and IPv6) HTTP proxy server as the default proxy.
endif::[]
ifndef::satellite[]

> * Optional: If you rely on content from IPv4-only networks and are deployed on IPv6-only, you can use an external dual stack HTTP proxy server.

Reviewer 4: ```suggestion
* If you rely on content from IPv4-only networks and are deployed on IPv6-only, you must use an external dual stack HTTP proxy.
```

question: I see this as conditional but not optional. If you consume content from an IPv4-only network, you _must_ use an HTTP proxy. What do you think?

Reviewer 2: Ooh, that's a good point. Thanks!

> In this setup, you must configure {Project} to use this dual-stack (supporting both IPv4 and IPv6) HTTP proxy server as the default proxy.

Reviewer 4: ```suggestion
In this setup, you must configure {Project} to use this dual-stack (supporting both IPv4 and IPv6) HTTP proxy as the default HTTP proxy.
```

endif::[]
+
For more information, see {InstallingServerDocURL}adding-a-default-http-proxy_{project-context}[Adding a Default HTTP Proxy to {Project}].

> Although {Project} provisioning templates include IPv6 support for PXE and HTTP (iPXE) provisioning, the only tested and certified provisioning workflow is the UEFI HTTP Boot provisioning.
> This limitation only relates to users who plan to use {Project} to provision hosts.

Reviewer 3: ```suggestion
```
I think that these lines no longer apply because of #3532 (TFTP ~ PXE). I think we can just remove them.

@ShimShtein Can you please confirm? Or is there another limitation?

Reviewer 5: I can confirm that I am working on enabling PXE provisioning in IPv6 networking.

As far as limitations go - I don't see anything yet

[id="ipv4-and-ipv6-requirements_{context}"]
= IPv4 and IPv6 requirements

You can install {Project} in an IPv4 network or in an IPv6 network.

ifdef::satellite[]
[NOTE]
====
Dual-stack {Project} installation that uses both IPv4 and IPv6 is not supported.
====
endif::[]

The following requirements apply to installations in an IPv4 network:

* An IPv6 loopback must be configured on the base system.

> ifndef::foreman-deb[]

Reviewer: This really is true any platform we deploy on AFAIK
